packages/foundry/contracts/hooks/NftCheckHook.sol

```diff
@@ -0,0 +1,351 @@
+            revert DoesNotOwnRequiredNFT(address(this), nftContract, nftId);
+        }
+
+        // Check if the linked token is one of the pool tokens
+        bool linkedTokenFound = false;
+        for (uint256 i = 0; i < tokenConfigs.length; i++) {
+            if (address(tokenConfigs[i].token) == linkedToken) {
+                linkedTokenFound = true;
+                break;
+            }
+        }
+        if (!linkedTokenFound) {
+            revert LinkedTokenNotInPool(linkedToken);
+        }
+        // Record the initial liquidity amounts
+        recordInitialLiquidity(tokenConfigs[0].token.balanceOf(poolAddress), tokenConfigs[1].token.balanceOf(poolAddress));
+
+        return true;
+    }
+
+    /// @inheritdoc IHooks
+    function getHookFlags() public pure override returns (HookFlags memory) {
+        HookFlags memory hookFlags;
+        hookFlags.shouldCallAfterRemoveLiquidity = true;
+        hookFlags.shouldCallBeforeInitialize = true;
+        hookFlags.shouldCallBeforeSwap = true;
+        hookFlags.shouldCallBeforeAddLiquidity = true;
+        return hookFlags;
+    }
+
+    /// @inheritdoc IHooks
+    function onAfterRemoveLiquidity(
+        address,
+        address pool,
+        RemoveLiquidityKind kind,
+        uint256,
+        uint256[] memory,
+        uint256[] memory amountsOutRaw,
+        uint256[] memory,
+        bytes memory
+    ) public override onlyVault returns (bool, uint256[] memory hookAdjustedAmountsOutRaw) {
+        
+        // Ensure the first depositor has the same amount of both tokens after removal (no rug pulling allowed)
+        if (msg.sender == owner()) {
+            IERC20[] memory poolTokens = _vault.getPoolTokens(pool);
+
+            uint256 currentToken1Amount = poolTokens[0].balanceOf(msg.sender);
+            uint256 currentToken2Amount = poolTokens[1].balanceOf(msg.sender);
+
+            if (currentToken1Amount < initialToken1Amount) {
+                revert InsufficientLiquidityToRemove(msg.sender, currentToken1Amount, initialToken1Amount);
+            }
+            if (currentToken2Amount < initialToken2Amount) {
+                revert InsufficientLiquidityToRemove(msg.sender, currentToken2Amount, initialToken2Amount);
+            }
+        }
+
+        // Our current architecture only supports fees on tokens. Since we must always respect exact `amountsOut`, and
+        // non-proportional remove liquidity operations would require taking fees in BPT, we only support proportional
+        // removeLiquidity.
+        if (kind != RemoveLiquidityKind.PROPORTIONAL) {
+            // Returning false will make the transaction revert, so the second argument does not matter.
+            return (false, amountsOutRaw);
+        }
+
+        IERC20[] memory tokens = _vault.getPoolTokens(pool);
+        uint256[] memory accruedFees = new uint256[](tokens.length);
+        hookAdjustedAmountsOutRaw = amountsOutRaw;
+
+        if (exitFeePercentage > 0) {
```

Comment:
Why is there an exit fee? In what instance would there be more than one LP in this pool? If there's only one LP, who are you donating fees too in the lines below?